app/models/product_nature_variant.rb

```diff
@@ -434,9 +434,8 @@ def current_stock
 
   # Return current quantity of all products link to the variant currently ordered or invoiced but not delivered
   def current_outgoing_stock_ordered_not_delivered
-    sales = Sale.where(state: %w[order invoice])
-    sale_items = SaleItem.where(variant_id: id, sale_id: sales.pluck(:id)).includes(:parcel_items).where(parcel_items: { sale_item_id: nil })
-    sale_items.map(&:quantity).compact.sum.to_f
+    sales_not_delivered = Sale.where(state: %w(order invoice)).includes(:parcels).where(parcels: { state: %w(draft ordered in_preparation prepared) })
+    SaleItem.where(variant_id: id, sale_id: sales_not_delivered.pluck(:id)).sum(:quantity).to_f
```

Comment:
@ionosphere said to me that the amount expected is the sum of ordered/invoiced quantity minus delivered quantity